### PR TITLE
Login providers should not be visible if the service is not configured

### LINKED ISSUE #13863: Facebook Login should not be visible if the service is not configured
Currently, if you enable the Facebook Login feature, a "Facebook" button will automatically show up for the user to login with even when the Facebook Integration settings is not set. 

The button should not show up unless the `FacebookLoginConfiguration` is set

### PR DESCRIPTION
**Pull request by lampersky**
fixes #13863

/cc @MikeAlhayek

### CODE REVIEW

**Comment by lampersky** [comment]
All tests pass in my environment, but it is failing here in ubuntu.

image

```
fail: Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware[1]
      An unhandled exception has occurred while executing the request.
      System.ArgumentException: An item with the same key has already been added. Key: PartName
         at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
         at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
         at OrchardCore.ContentManagement.GraphQL.FieldTypeExtensions.WithMetaData(FieldType fieldType, String name, Object value) in /_/src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Extensions/FieldTypeExtensions.cs:line [26](https://github.com/OrchardCMS/OrchardCore/actions/runs/7862582347/job/21452217647?pr=15305#step:6:27)
```

**Comment by MikeAlhayek** [comment]
@lampersky can you please open an new issue with that? I think this is a new introduced by the last update we had for graphQL. It should be addressed.  I seen it happen with other tests also

File: src/OrchardCore.Modules/OrchardCore.Facebook/Login/Configuration/FacebookLoginConfiguration.cs [hunk 1]
```diff
@@ -43,6 +43,11 @@ public void Configure(AuthenticationOptions options)
                 return;
             }
 
+            if (string.IsNullOrWhiteSpace(_facebookSettings.AppId) || string.IsNullOrWhiteSpace(_facebookSettings.AppSecret))
+            {
+                return;
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Facebook/Login/Configuration/FacebookLoginConfiguration.cs:48]: Here you can add a log with something like.

The Facebook login provider is enabled but not configured.

File: src/OrchardCore.Modules/OrchardCore.GitHub/Configuration/GithubOptionsConfiguration.cs [hunk 2]
```diff
@@ -32,6 +32,12 @@ public void Configure(AuthenticationOptions options)
                 return;
             }
 
+            if (string.IsNullOrWhiteSpace(_gitHubAuthenticationSettings.ClientID) ||
+                string.IsNullOrWhiteSpace(_gitHubAuthenticationSettings.ClientSecret))
+            {
+                return;
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.GitHub/Configuration/GithubOptionsConfiguration.cs:38]: Add a log here as well

File: src/OrchardCore.Modules/OrchardCore.Google/Authentication/Configuration/GoogleOptionsConfiguration.cs [hunk 3]
```diff
@@ -12,23 +12,29 @@ public class GoogleOptionsConfiguration :
...
             _dataProtectionProvider = dataProtectionProvider;
             _logger = logger;
         }
 
         public void Configure(AuthenticationOptions options)
         {
-            if (_gitHubAuthenticationSettings == null)
+            if (_googleAuthenticationSettings == null)
+            {
+                return;
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Google/Authentication/Configuration/GoogleOptionsConfiguration.cs:33]: Log

File: src/OrchardCore.Modules/OrchardCore.Google/Authentication/Configuration/GoogleOptionsConfiguration.cs [hunk 4]
```diff
@@ -12,23 +12,29 @@ public class GoogleOptionsConfiguration :
...
+            }
+
+            if (string.IsNullOrWhiteSpace(_googleAuthenticationSettings.ClientID) ||
+                string.IsNullOrWhiteSpace(_googleAuthenticationSettings.ClientSecret))
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Google/Authentication/Configuration/GoogleOptionsConfiguration.cs:37]: Log before the return

File: src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/AzureADOptionsConfiguration.cs [hunk 5]
```diff
@@ -31,6 +31,11 @@ public void Configure(AuthenticationOptions options)
                 return;
             }
 
+            if (string.IsNullOrWhiteSpace(settings.AppId) || string.IsNullOrWhiteSpace(settings.TenantId))
+            {
+                return;
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/AzureADOptionsConfiguration.cs:36]: Log

File: src/OrchardCore.Modules/OrchardCore.Twitter/Signin/Configuration/TwitterOptionsConfiguration.cs [hunk 6]
```diff
@@ -59,6 +59,12 @@ public void Configure(AuthenticationOptions options)
                 return;
             }
 
+            if (string.IsNullOrWhiteSpace(settings.Item1.ConsumerKey) ||
+                string.IsNullOrWhiteSpace(settings.Item1.ConsumerSecret))
+            {
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Twitter/Signin/Configuration/TwitterOptionsConfiguration.cs:64]: Log

**Review by MikeAlhayek — CHANGES_REQUESTED** [review]
LGTM! I think some add these logs will help

**Review by MikeAlhayek — APPROVED** [review]

**Pull request merged**